### DISCUSSION TITLE
`next-env.d.ts` adding a new line on each `next dev` / `next build`

### DISCUSSION THREAD

**Discussion by omrimor** [vercel/next.js#66994] [Help]
### Summary

We are two developers working on the same codebase.
When I run `next dev` a line in `next-env.d.ts` gets added for me locally but does not happen on the other developer's machine.

I cleaned `node_modules`, tried to clean `yarn cache` but no luck.

```diff
/// <reference types="next" />
/// <reference types="next/image-types/global" />
+ /// <reference types="next/navigation-types/compat/navigation" />

// NOTE: This file should not be edited
// see https://nextjs.org/docs/basic-features/typescript for more information.
```

**Comment by glifchits** [comment]
Our team is experiencing this as well. What accounts for this line being added when we're all working on the same codebase and how should we make it consistent?

According to [this related thread](https://github.com/vercel/next.js/discussions/47010#discussion-4949792), there may have been some advice in the docs to `.gitignore` the `next-env.d.ts` file, but I cannot find that quote anymore in the [Next TS docs](https://nextjs.org/docs/app/building-your-application/configuring/typescript) or anywhere else. 
> A file named next-env.d.ts will be created at the root of your project. This file ensures Next.js types are picked up by the TypeScript compiler. You should not remove it or edit it as it can change at any time. **This file should not be committed and should be ignored by version control (e.g. inside your .gitignore file).**

**Comment by omrimor** [comment]
Actually I think I found the issue in my case.
I had a leftover empty `app` folder from a different branch and that folder was left empty when I switched back and forth between work branches.

I think this is what cause the `next-env.d.ts` to update all the time, it picked up that folder and added the types for the app router navigation package. 

Once I deleted the ghost folder, It stoped updating this file

**Reply by glifchits** [reply]
Ahhhh amazing, thank you for reporting this find!

**Reply by jjablonski-it** [reply]
I had a ghost `pages` directory and removing it solved the issue for me. Thank you @omrimor!

**Reply by kyleleow** [reply]
for my case, it was the empty `pages` directory! Thank you!

**Reply by VolkerFried** [reply]
also empty pages directory! thanks!!!

**Comment by padcmoi** [comment]
nextjs version used: 15.2.3

I can confirm that this error does occur when a phantom folder is created somewhere. Next.js then behaves in a very strange and sometimes quite frustrating way, as the errors it produces seem completely unrelated.

Take the following example:
In an application using the App Router, add a pages folder (as if you were using the old routing system).
When you build the app, you will see the following line appear in `next-env.d.ts`:

```
/// <reference types="next/navigation-types/compat/navigation" />
```

But that is not all. You may also notice other strange behaviors, such as:

- `const pathname: string | null`
- `const searchParams: ReadonlyURLSearchParams | null`

These anomalies can make you think you missed an error, even after thoroughly testing before committing.
It can really drive you crazy!

**Reply by kmaid** [reply]
me too same version of next. I am wondering if its because I have sentry installed?

**Comment by SanskarDahiya** [comment]
I can confirm that this error does occur when a phantom folder is created somewhere
either remove app folder or remove pages folder.
